src/main/scala/firrtl/backends/experimental/smt/SMTExpr.scala

```diff
@@ -5,9 +5,20 @@
 
 package firrtl.backends.experimental.smt
 
-private sealed trait SMTExpr { def children: List[SMTExpr] }
-private sealed trait SMTSymbol extends SMTExpr with SMTNullaryExpr { val name: String }
+/** base trait for all SMT expressions */
+private sealed trait SMTExpr extends SMTFunctionArg {
+  def tpe:      SMTType
+  def children: List[SMTExpr]
+}
+private sealed trait SMTSymbol extends SMTExpr with SMTNullaryExpr {
+  val name: String
```

Comment:
Nit but generally virtual methods in traits should always be `def`--it doesn't preclude using a `val` in the concrete instances.

Reply:
fixed